sql3/test/defs/defs_groupby.go

```diff
@@ -323,6 +324,40 @@ var groupBySetDistinctTests = TableTest{
 			),
 			Compare: CompareExactUnordered,
 		},
+		{
+			SQLs: sqls(
+				"select distinct ids1, ss1 from groupby_set_test",
+			),
+			ExpHdrs: hdrs(
+				hdr("ids1", fldTypeIDSet),
+				hdr("ss1", fldTypeStringSet),
+			),
+			ExpRows: rows(
+				row([]int64{1, 2}, []string{"a", "b"}),
+				row([]int64{3, 4}, []string{"d", "e"}),
+				row([]int64{1, 4}, []string{"a", "d"}),
+				row([]int64{2, 3}, []string{"b", "c"}),
+			),
+			Compare:        CompareExactUnordered,
+			SortStringKeys: true,
+		},
+		{
+			SQLs: sqls(
+				"select distinct ids1, ss1 from groupby_set_test with (flatten(ids1))",
+			),
+			ExpHdrs: hdrs(
+				hdr("ids1", fldTypeIDSet),
+				hdr("ss1", fldTypeStringSet),
+			),
+			ExpRows: rows(
```

Comment:
This doesn't look like `ids1` got flattened?

I am ... not actually sure what I expect here. I think I sort of expect the id set column values to all be 1, 2, 3, or 4, rather than sets, because they're being flattened. And I think that implies something like

1, {"a", "b"}
2, {"a", "b"}
3, {"d", "e"}
4, {"d", "e"}

etcetera? but now i'm not sure.